src/libsyncengine/propagation/executor/executorworker.cpp

```diff
@@ -504,46 +469,28 @@ bool ExecutorWorker::checkAlreadyExcluded(const SyncPath &absolutePath, const No
     }
 
     if (!alreadyExist) {
-        return true;
-    }
-
-    // The item already exists, exclude it
-    exitCode = PlatformInconsistencyCheckerUtility::renameLocalFile(absolutePath,
-                                                                    PlatformInconsistencyCheckerUtility::SuffixType::Blacklisted);
-    if (exitCode != ExitCode::Ok) {
-        LOGW_SYNCPAL_WARN(_logger, L"Failed to rename file: " << Utility::formatSyncPath(absolutePath).c_str());
-        _executorExitCode = exitCode;
-        _executorExitCause = ExitCause::Unknown;
-        return false;
+        return ExitCode::Ok;
     }
```

Comment:
So we do not blacklist item anymore in that case? Is it done somewhere else?
If not, is there a chance of infinite sync loop?

Reply:
It is handled somewhere else.

This function (`checkAlreadyExcluded`) is called in `ExecutorWorker::handleCreateOp`:

```cpp
if (ExitInfo exitInfoCheckAlreadyExcluded =
            checkAlreadyExcluded(absoluteLocalFilePath, createDirJob->parentDirId());
    !exitInfoCheckAlreadyExcluded) {
    LOG_SYNCPAL_WARN(_logger, "Error in ExecutorWorker::checkAlreadyExcluded");
    return exitInfoCheckAlreadyExcluded;
}
```

If the file already exists, `handleCreateOp` will return with `ExitCode::DataError` and `ExitCause::FileAlreadyExist`.

In `execute`, we then have:

```cpp
ExecutorWorker::execute() {
...
  case OperationType::Create: {
      executorExitInfo = handleCreateOp(syncOp, job, ignored);
      break;
  }
  ...
  ...
  if (!executorExitInfo) {
      executorExitInfo = handleExecutorError(syncOp, executorExitInfo);
  ...
  }
}
```

The file already exists case is defined in `handleExecutorError`:

```cpp
ExecutorWorker::handleExecutorError(SyncOpPtr syncOp, ExitInfo opsExitInfo) {
  ...
  // Handle specific errors
  switch (static_cast<int>(opsExitInfo)) {
      case static_cast<int>(ExitInfo(ExitCode::SystemError, ExitCause::FileAccessError)):
      case static_cast<int>(ExitInfo(ExitCode::SystemError, ExitCause::MoveToTrashFailed)): {
          return handleOpsFileAccessError(syncOp, opsExitInfo);
      }
      case static_cast<int>(ExitInfo(ExitCode::BackError, ExitCause::FileAlreadyExist)):
      case static_cast<int>(ExitInfo(ExitCode::DataError, ExitCause::FileAlreadyExist)): {
          return handleOpsAlreadyExistError(syncOp, opsExitInfo);
      }
      default: {
          break;
      }
  };
```

See `executorworker.cpp` line 2598 for the detailed handling of `AlreadyExistError`. In summary, if the file is not in the temporary blacklist on the local side, it will be renamed with the suffix `_blacklisted_;` otherwise, the remote file will also be added to the temporary blacklist.

Let me know if you see a flaw in this design.

Reply:
Sounds good, thx